src/Sulu/Bundle/TestBundle/Testing/TestUserTrait.php

```diff
@@ -11,19 +11,32 @@
 
 namespace Sulu\Bundle\TestBundle\Testing;
 
+use Sulu\Bundle\AdminBundle\Admin\Admin;
 use Sulu\Bundle\SecurityBundle\Entity\User;
+use Sulu\Component\Security\Authentication\RoleInterface;
 
 trait TestUserTrait
 {
     /**
-     * Return the test user (which is provided / created
+     * Return a test user (which is provided / created
      * by the test_user_provider in this Bundle at runtime).
      *
      * @return User
      */
-    protected static function getTestUser()
+    protected static function getTestUser(string $username = TestUserProvider::TEST_USER_USERNAME)
```

Comment:
I think I would leave that file untouched, and create the entities in the classes as necessary, probably creating a `createRole` and/or `createUser` function as required. Then we don't have any side effects among different test cases, which are hard to fix.